src/Interpreters/Context.cpp

```diff
@@ -933,6 +933,17 @@ bool Context::hasScalar(const String & name) const
 }
 
 
+void Context::addQueryAccessInfo(const String & database_name, const String & table_name, const Names & column_names)
+{
+    assert(global_context != this || getApplicationType() == ApplicationType::LOCAL);
+    auto lock = getLock();
+    query_access_info.databases.emplace(database_name);
+    query_access_info.tables.emplace(table_name);
+    for (const auto & column_name : column_names)
+        query_access_info.columns.emplace(table_name + "." + column_name);
```

Comment:
backquoteIfNeed.

Reply:
Done.

Reply:
Better to apply backQuoteIfNeed to all parts.

Reply:
BTW, why we don't prepend database name also?

Reply:
Because it's already full table name passed in.. I'll change the name of the argument to indicate that.